docs/intro-to-dune/index.md

```diff
@@ -0,0 +1,284 @@
+[dirs](https://dune.readthedocs.io/en/stable/dune-files.html#dirs), which tells
+Dune which directories to include in the build. Note that the stanzas accepted
+in `dune` files are not the same as the ones accepted by `dune-project`. See all
+possible `dune` stanzas in Dune's [Stanza
+Reference](https://dune.readthedocs.io/en/stable/dune-files.html#dune).
+
+## `melange.emit` stanza
+
+Let's now take a look at the `src/dune` file:
+
+<<< @/../src/counter/dune{clj}
+
+The main stanza of interest for us is
+[melange.emit](https://dune.readthedocs.io/en/stable/melange.html#melange-emit),
+which tells Dune to turn our OCaml files into JavaScript files. The fields we
+give to `melange.emit` here are `target`, `libraries`, `preprocess`, and
+`module_systems`, which are ones that we need to use for pretty much every
+Melange project.
+
+A note about the `target` field: It is referenced in the `serve` npm script,
+whose command is
+
+```bash
+webpack serve --open --mode development --entry ./_build/default/src/output/src/Index.js
+```
+
+If you change the value of `target` to, say, `stuff`, the value of the `--entry`
+option must be changed to
+
+```
+./_build/default/src/stuff/src/Index.js
+```
+
+::: info
+In this dune file, we're only concerned with building JavaScript to run in the
+browser, but if we also wanted to build JavaScript to run on Node, we'd include
+another `melange.emit` stanza specifically for that. See
+[melange-opam-template](https://github.com/melange-re/melange-opam-template/blob/main/src/dune)
+for an example of building for Node.
+:::
+
+## Counter app directory
+
+Create a new directory in `src` called `counter`, which will contain a new app
+that only renders the [Counter component](/counter/). Then make sure this new
+directory contains:
+
+- `Counter.re` from `src`
+- `dune` file from `src`
+- `Index.re` to render the app to the DOM
+
+  <<< @/../src/counter/Index.re
+- `Makefile` to serve the app using webpack dev server
+
+  <<< @/../src/counter/Makefile{make}
+
+After you've added those files, your `src/counter` directory should look like
+this:
+
+```
+src/counter
+├─ Counter.re
+├─ dune
+├─ Index.re
+└─ Makefile
+```
+
+## Why Makefile
+
+You might be wondering why we're using a `Makefile` instead of adding another
+npm script. In the world of OCaml, usage of [the Make build automation
+tool](https://en.wikipedia.org/wiki/Make_(software)) is pretty much ubiquitous,
+and this is not likely to change. So let's take a short detour to explain the
+`Makefile` we just added.
+
+## Anatomy of Makefile
+
+In case you're not familiar with Make, here's a breakdown of
+`src/counter/Makefile`:
+
+<<< @/../src/counter/Makefile{make}
+
+- There's only a single rule here, and its *target* is `serve`.
+- `.PHONY: serve` is used to indicate that `serve` is a [phony
+  target](https://makefiletutorial.com/#phony), meaning `serve` is not a file we
+  want to generate but just a label for an action. In a simple `Makefile` like
```

Comment:
If I cover dune aliases in the book, I assume it will be much farther down the line, when I introduce a small OCaml server app. So I think I'll skip the reference for now.

Reply:
Sounds good.